tools/mo/openvino/tools/mo/convert_impl.py

```diff
@@ -432,23 +434,38 @@ def read_model(path_to_xml):
     mean_data = deepcopy(graph.graph['mf']) if 'mf' in graph.graph else None
     input_names = deepcopy(graph.graph['input_names']) if 'input_names' in graph.graph else []
 
-    prepare_emit_ir(graph=graph,
-                    data_type=graph.graph['cmd_params'].data_type,
-                    output_dir=argv.output_dir,
-                    output_model_name=argv.model_name,
-                    mean_data=mean_data,
-                    input_names=input_names,
-                    meta_info=non_default_params,
-                    use_temporary_path=True)
-
-    # This graph cleanup is required to avoid double memory consumption
-    graph.clear()
-
     output_dir = argv.output_dir if argv.output_dir != '.' else os.getcwd()
     orig_model_name = os.path.normpath(os.path.join(output_dir, argv.model_name))
 
-    fem = FrontEndManager()
-    func = read_model(orig_model_name + "_tmp.xml")
+    try:
+        prepare_emit_ir(graph=graph,
+                        data_type=graph.graph['cmd_params'].data_type,
+                        output_dir=argv.output_dir,
+                        output_model_name=argv.model_name,
+                        mean_data=mean_data,
+                        input_names=input_names,
+                        meta_info=non_default_params,
+                        use_temporary_path=True)
+
+        fem = FrontEndManager()
+        func = read_model(fem, orig_model_name + "_tmp.xml")
+    except Error as err:
+        raise Error('Error occurred during serialization of temporary IR: {}'.format(
+            str(err),
+        )) from err
+    except Exception as err:
+        raise Exception('Exception occurred during serialization of temporary IR: {}'.format(
+            str(err),
+        )) from err
```

Comment:
exception can occur not only during serialization. It can occur during reading by IR frontend. Please correct these blocks

Reply:
added separated catch to distinguish Exceptions while serializing from errors while loading